### ISSUE TITLE
Stack overflow in Jest still exists

### ISSUE THREAD

**Issue by pawsong** [Automattic/mongoose#5513]
I know this issue has already been discussed in https://github.com/Automattic/mongoose/issues/4943, but I am still facing this issue :(

Here is a repro script with latest packages:

- https://github.com/pawsong/mongoose-reproduce-4943
- mongoose v4.11.4
- jest v20.0.4
- node v7.10.0

**Comment by vkarpov15** [comment]
As a workaround, do this:

```javascript
/**
 * @jest-environment node
 */

const mongoose = require('mongoose');
const TestModel = require('../Test');

describe('Test', () => {
  beforeAll(() => {
    mongoose.connect('mongodb://localhost/test');
  });

  it('should overflow the stack', () => {
    const doc = new TestModel({
      nested: {
        arr: [
          { str: 'blabla' },
        ],
      },
    });
    return doc.save();
  });
});
```

We suspect this issue is due to the fact that jest uses jsdom, which has a conflict with mongoose that we have yet to identify. If you run it using the [node test environment](https://facebook.github.io/jest/docs/configuration.html) it'll work just fine. However, I must recommend you avoid using jest and Facebook OSS projects in general, they tend to be littered with childish bugs and cause way more problems than they solve.

**Comment by vkarpov15** [comment]
Found the bug and fixed it. Thanks for the detailed repro, fix will be in 4.11.6. However, please make sure you use the node test environment, because otherwise you'll be testing mongoose's browser side code and won't be able to actually save documents.

**Comment by pawsong** [comment]
@vkarpov15 Thank you for your feedback!
